File: frontend/src/types/template.ts
import type { JobType } from "../types/api";

export interface JobTemplate {
  id: string;
  name: string;
  description: string;
  job_type: JobType;
  source_system: string;
  target_system: string;
  default_job_name: string;
  schedule_cron: string | null;
  config: Record<string, unknown>;
  integration_mode: "simulated" | "integrated" | string;
}

File: backend/app/schemas/template.py
from pydantic import BaseModel, Field

from app.models.job import JobType


class JobTemplateRead(BaseModel):
    id: str
    name: str
    description: str
    job_type: JobType
    source_system: str
    target_system: str
    default_job_name: str
    schedule_cron: str | None = None
    config: dict = Field(default_factory=dict)
    integration_mode: str = Field(
        default="simulated",
        description="simulated = metadata only; integrated = live connector (post-MVP)",
    )
